mdbenchmark/analyze.py

```diff
@@ -182,19 +103,27 @@ def analyze(directory, plot, ncores, output_name):
     df.to_csv(output_name)
 
     if plot:
+        console.warn(
+            'This feature is now outdated.'
+            'Please use mdbenchmark plot to access all plotting features'
+            'Future versions will not support this feature anymore.')
+
+        fig = Figure()
+        FigureCanvas(fig)
+        ax = fig.add_subplot(111)
+
         df = pd.read_csv(output_name)
 
-        # We only support plotting of benchmark systems from equal hosts /
-        # with equal settings
-        uniqueness = df.apply(lambda x: x.nunique())
-        if uniqueness['gromacs'] > 1 or uniqueness['host'] > 1:
-            console.error(
-                'Cannot plot benchmarks for more than one GROMACS module '
-                'and/or host.')
+        df_sel = 'nodes'
+        gb = df.groupby(['gpu', 'module', 'host'])
+        groupby = ['gpu', 'module', 'host']
+        for key, df in gb:
+            label = ' '.join(['{}={}'.format(n, v) for n, v in zip(groupby, key)])
+            plot_line(df=df, df_sel=df_sel, ax=ax, label=label)
 
-        # Fail if we have no values at all. This should be some edge case when
-        # a user fumbles around with the datreant categories
-        if df['gpu'].empty and df[~df['gpu']].empty:
-            console.error('There is no data to plot.')
+        ax.set_xlabel('Number of Nodes')
+        ax.set_ylabel('Performance [ns/day]')
```

Comment:
ax.set(xlabel=' ... ', ylabel='...')

Reply:
I don't see why using the current approach is bad. If we ever change the labels to something longer, we may end up with two lines anyway.